components/transactions/TransactionDetails.js

```diff
@@ -48,28 +50,34 @@ const DetailsContainer = styled(Flex)`
   }
 `;
 
-const TransactionDetails = ({
-  displayActions,
-  id,
-  type,
-  isRefunded,
-  toAccount,
-  fromAccount,
-  uuid,
-  platformFee,
-  hostFee,
-  paymentMethod,
-  paymentProcessorFee,
-  amount,
-  netAmount,
-  permissions,
-  order,
-  expense,
-}) => {
+const TransactionDetails = ({ displayActions, transaction, isRoot, isHostAdmin }) => {
   const intl = useIntl();
   const isCredit = type === TransactionTypes.CREDIT;
   const hasOrder = order !== null;
   const { loading: loadingInvoice, callWith: downloadInvoiceWith } = useAsyncCall(saveInvoice);
+  const {
+    id,
+    type,
+    isRefunded,
+    toAccount,
+    fromAccount,
+    uuid,
+    platformFee,
+    hostFee,
+    paymentMethod,
+    paymentProcessorFee,
+    amount,
+    netAmount,
+    permissions,
+    order,
+    expense,
+    isRejected,
+    createdAt,
+  } = transaction;
+  const timeLimit = moment().subtract(30, 'd');
+  const createdAtMoment = moment(createdAt);
+  const transactionIsOlderThanThirtyDays = createdAtMoment < timeLimit;
```

Comment:
We're [getting rid of moment](https://github.com/opencollective/opencollective/issues/2446), let's avoid using it for new devs. Also it may be more explicit to rename `timeLimit` => `timeLimitForRefunds`.

I would have personally implemented this logic in `permissions.canRefund` to avoid code duplication and because it's easier to tests permissions when they're centralized, but it's not a blocker.

Reply:
Will update in next iteration

Reply:
Since we're not merging til later today will add this too